tests/GraphNode/GraphNodeFactoryTest.php

```diff
@@ -258,6 +258,27 @@ public function testAGraphNodeWithARootDataKeyWillBeCastAsAGraphNode()
         ], $graphData);
     }
 
+    public function testAGraphNodeWithARootDataKeyWillConserveId()
+    {
+        $data = json_encode([
+            'id' => '123',
```

Comment:
the should be a test without this field, showing that the node becomes the data field values (you could use the @tolbon value in https://github.com/facebook/php-graph-sdk/pull/817#issuecomment-438615073). And another test ensuring that nothing change when there are other fields next to the data field (using your test data here)